### PR TITLE
Implement match chat redirection

### PR DESCRIPTION
**Pull request by vianxnguyen**
## Overview

Implement chat redirection for matched pair, added current user information to other profile screens.

## Changes Made

- Added reach out button to chat with matched pair
- Passed current user information to other user profiles, in case we need to implement chatting with other users.

## Test Coverage

Tested on device.

### CODE REVIEW

File: Pear/Controller/CommunityViewController.swift [hunk 1]
```diff
@@ -16,9 +16,20 @@ class CommunityViewController: UIViewController {
         fadeColor: .backgroundLightGreen)
     private let searchBar = UISearchBar()
     private let spinner = UIActivityIndicatorView(style: .medium)
-
+    
     // MARK: - Private Data Vars
     private var users: [CommunityUser] = []
+    private var currentUser : UserV2
```
Comment (ReadeP) [inline Pear/Controller/CommunityViewController.swift:22]: no space between currentUser and :

File: Pear/Controller/ChatViewController.swift [hunk 2]
```diff
@@ -301,7 +301,7 @@ extension ChatViewController: UITableViewDataSource {
         let message = groupedMessagesByDate[indexPath.section][indexPath.row]
         cell.configure(for: message, user: currentUser, pair: messageUser)
         cell.viewProfile = {
-            self.navigationController?.pushViewController(ProfileViewController(userId: self.messageUser.id), animated: true)
+            self.navigationController?.pushViewController(ProfileViewController(user: self.currentUser, otherUserId: self.messageUser.id), animated: true)
```
Comment (ReadeP) [inline Pear/Controller/ChatViewController.swift:304]: Remove "self" before currentUser and messageUser

File: Pear/Profile/MatchProfileViewController.swift [hunk 3]
```diff
@@ -107,11 +116,24 @@ class MatchProfileViewController: UIViewController {
...
+
     }
 
     @objc func backPressed() {
         navigationController?.popViewController(animated: true)
     }
+    
+    @objc func reachOutButtonPressed() {
+        if let matchedUser = match {
+        navigationController?.pushViewController(ChatViewController(messageUser: matchedUser, currentUser: user), animated: true)
```
Comment (ReadeP) [inline Pear/Profile/MatchProfileViewController.swift:134]: Indent this line

File: Pear/Profile/ProfileViewController.swift [hunk 4]
```diff
@@ -13,13 +13,16 @@ class ProfileViewController: UIViewController {
...
     private let profileTableView = UITableView(frame: .zero, style: .plain)
-    private var userId: Int
+    private var otherUserId: Int
 
-    init(userId: Int) {
-        self.userId = userId
+    init(user: UserV2, otherUserId: Int) {
+        self.currentUser = user
+        self.otherUserId = otherUserId
+        print(otherUserId)
```
Comment (ReadeP) [inline Pear/Profile/ProfileViewController.swift:25]: delete this print statement

File: Pear/Controller/CommunityViewController.swift [hunk 5]
```diff
@@ -16,9 +16,20 @@ class CommunityViewController: UIViewController {
         fadeColor: .backgroundLightGreen)
     private let searchBar = UISearchBar()
     private let spinner = UIActivityIndicatorView(style: .medium)
-
+
```
Comment (danielVebman) [inline Pear/Controller/CommunityViewController.swift:19]: Delete whitespace

File: Pear/Controller/CommunityViewController.swift [hunk 6]
```diff
@@ -16,9 +16,20 @@ class CommunityViewController: UIViewController {
         fadeColor: .backgroundLightGreen)
     private let searchBar = UISearchBar()
     private let spinner = UIActivityIndicatorView(style: .medium)
-
+    
     // MARK: - Private Data Vars
     private var users: [CommunityUser] = []
+    private var currentUser: UserV2
```
Comment (danielVebman) [inline Pear/Controller/CommunityViewController.swift:22]: If this isn't modified, make it `private let`

File: Pear/Controller/TabPageViewController.swift [hunk 7]
```diff
@@ -15,7 +15,7 @@ protocol TabDelegate: AnyObject {
 class TabPageViewController: UIPageViewController {
 
     // MARK: - Private View Vars
-    private let communityViewController = CommunityViewController()
+    private let communityViewController : UIViewController
```
Comment (danielVebman) [inline Pear/Controller/TabPageViewController.swift:18]: Delete space before `:`, i.e.

```suggestion
    private let communityViewController: UIViewController
```

File: Pear/Profile/MatchProfileViewController.swift [hunk 8]
```diff
@@ -107,11 +116,24 @@ class MatchProfileViewController: UIViewController {
                 make.edges.equalToSuperview()
             }
         }
+        
+        reachOutButton.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.size.equalTo(Constants.Onboarding.largeButtonSize)
+            make.bottom.equalTo(view.safeAreaLayoutGuide).inset(Constants.Onboarding.nextBottomPadding)
```
Comment (danielVebman) [inline Pear/Profile/MatchProfileViewController.swift:123]: If this is very long, break up onto multiple lines.

File: Pear/Profile/MatchProfileViewController.swift [hunk 9]
```diff
@@ -107,11 +116,24 @@ class MatchProfileViewController: UIViewController {
...
+        }
+
```
Comment (danielVebman) [inline Pear/Profile/MatchProfileViewController.swift:125]: Delete newline

File: Pear/Profile/MatchProfileViewController.swift [hunk 10]
```diff
@@ -107,11 +116,24 @@ class MatchProfileViewController: UIViewController {
...
     }
 
     @objc func backPressed() {
         navigationController?.popViewController(animated: true)
     }
+    
+    @objc func reachOutButtonPressed() {
+        if let matchedUser = match {
+            navigationController?.pushViewController(ChatViewController(messageUser: matchedUser, currentUser: user), animated: true)
+            }
```
Comment (danielVebman) [inline Pear/Profile/MatchProfileViewController.swift:135]: Indentation off

File: Pear/Profile/MatchProfileViewController.swift [hunk 11]
```diff
@@ -107,11 +116,24 @@ class MatchProfileViewController: UIViewController {
...
+
     }
 
     @objc func backPressed() {
         navigationController?.popViewController(animated: true)
     }
+    
+    @objc func reachOutButtonPressed() {
+        if let matchedUser = match {
+            navigationController?.pushViewController(ChatViewController(messageUser: matchedUser, currentUser: user), animated: true)
```
Comment (danielVebman) [inline Pear/Profile/MatchProfileViewController.swift:134]: Break up onto multiple lines. E.g.

```suggestion
            navigationController?.pushViewController(
                ChatViewController(
                    messageUser: matchedUser, 
                    currentUser: user
                ), 
                animated: true
            )
```

File: Pear/Profile/ProfileViewController.swift [hunk 12]
```diff
@@ -13,13 +13,15 @@ class ProfileViewController: UIViewController {
 
     // MARK: - Private View Vars
     private let backButton = UIButton()
-    private var user: UserV2?
+    private var currentUser: UserV2
+    private var otherUser: UserV2?
     private var profileSections = [ProfileSectionType]()
     private let profileTableView = UITableView(frame: .zero, style: .plain)
-    private var userId: Int
+    private var otherUserId: Int
```
Comment (danielVebman) [inline Pear/Profile/ProfileViewController.swift:20]: Why not let?

File: Pear/Profile/ProfileViewController.swift [hunk 13]
```diff
@@ -61,20 +63,20 @@ class ProfileViewController: UIViewController {
         profileTableView.contentInset = UIEdgeInsets(top: 40, left: 0, bottom: 40, right: 0)
         profileTableView.showsVerticalScrollIndicator = false
         view.addSubview(profileTableView)
-
+
```
Comment (danielVebman) [inline Pear/Profile/ProfileViewController.swift:66]: delete whitespace

File: Pear/Profile/ProfileViewController.swift [hunk 14]
```diff
@@ -13,13 +13,15 @@ class ProfileViewController: UIViewController {
 
     // MARK: - Private View Vars
     private let backButton = UIButton()
-    private var user: UserV2?
+    private var currentUser: UserV2
+    private var otherUser: UserV2?
```
Comment (danielVebman) [inline Pear/Profile/ProfileViewController.swift:17]: Having this optional variable and also having `otherUserId` is redundant. What if you made an enum with two states? 

```
private enum ProfileViewControllerUser {
    case userId(String)
    case user(UserV2)
}
```

Maybe this is unnecessary. Let me know your thoughts.

**Review by danielVebman — CHANGES_REQUESTED** [review]
Great first PR Vivian! Let me know if you have any questions about my comments. Re-request my review once you finish making changes, and make sure to give most of your reviewers a chance to review your code before you merge!

File: Pear/Profile/ProfileViewController.swift (hunk 12 above)
Comment (danielVebman) [reply Pear/Profile/ProfileViewController.swift:20]: That doesn't matter. You can say

```
class X {
    private let y: Int
    init(y: Int) { self.y = y }
}
```

File: Pear/Profile/ProfileViewController.swift [hunk 15]
```diff
@@ -63,16 +62,12 @@ class ProfileViewController: UIViewController {
...
         profileTableView.showsVerticalScrollIndicator = false
         view.addSubview(profileTableView)
-        
         setupConstraints()
-        
-        getUser()
     }
 
-    private func getUser() {
+    private func setOtherUser(otherUserId: Int) {
```
Comment (danielVebman) [inline Pear/Profile/ProfileViewController.swift:68]: I'd call this `getOtherUser`

**Review by danielVebman — APPROVED** [review]
lgtm!

**Comment by danielVebman** [comment]
It would be ideal if you included screenshots and/or screen recordings in your PR description.

**Pull request merged**